packages/@ember/-internals/bootstrap/index.js

```diff
@@ -28,4 +29,11 @@ import { context } from '@ember/-internals/environment';
     // eslint-disable-next-line no-undef
     module.exports = Ember = require('ember').default;
   }
+
+  if (ENV._JQUERY_INTEGRATION) {
+    // eslint-disable-next-line no-undef
+    define('jquery', [], () => {
+      return jQuery;
+    });
```

Comment:
Yeah, it does, but I couldn't really think of a better way to add this. `@ember/jquery` adds its own shim as is, and if the integration is disabled, it should not exist and throw an error. So it has to be dynamically defined to match the existing behavior.

Reply:
Why does that mean it has to be dynamically defined (vs creating `packages/jquery/index.js`)? With the modules API polyfill, `import jQuery from 'jquery'` is transformed into `Ember.$` unless `@ember/jquery` is present (see https://github.com/babel/ember-cli-babel/blob/5eafd6baad763786246e06f862704a770be6ffc9/lib/ember-plugins.js#L74-L81).

I would think we could define a `jquery` module when `@ember/jquery` is not present, and that module would assert / deprecate as appropriate...

Reply:
right, that's what this does, it just does it dynamically in the browser vs the more complicated approach of doing it in `ember-cli`/`babel`. We could update it there, it just feels a bit more complicated

Reply:
I don't understand. If you put a file in `packages/jquery/index.js` won't it "just work"? Are you concerned that you'd have to **conditionally** exclude that file?

Reply:
Since this is effectively "throw away code" (meaning we don't care about it in ember-source@4.0.0), I'd move this out into a separate vendor file that we `app.import` conditionally (based on presence of `@ember/jquery` and/or the optional feature flag setting).

Reply:
Ok, updated. I didn't add the check for `@ember/jquery` since it seems to work without it, the module gets defined twice is all but that doesn't throw.